### ISSUE TITLE
Feature Request: Add `clone()` method and immutable API for React compatibility

### ISSUE THREAD

**Issue by MOhhh-ok** [ggaabe/rrule-temporal#91]
## Description
Currently, `RRuleTemporal` doesn't provide a way to create a modified copy of an existing rule instance. The `options()` method returns a reference to the internal options object, which can be mutated directly, but this approach has several issues:

1. **No way to clone an instance**: There's no built-in method to create a copy of a rule with modified options
2. **React compatibility issues**: Direct mutation doesn't trigger re-renders in React's state management
3. **Breaks immutability patterns**: Goes against modern JavaScript best practices

## Use Case Example (React)
```typescript
const [rule, setRule] = useState(
  new RRuleTemporal({
    freq: "MONTHLY",
    byMonthDay: [1],
    dtstart,
    count: 3,
  })
);

// This doesn't work - mutates the object but React won't re-render
rule.options().byMonthDay = [3];

// Current workaround - verbose and error-prone
setRule(new RRuleTemporal({
  ...rule.options(),
  byMonthDay: [3],
}));
```

### Option 1: Add a `clone()` method
```typescript
const rule2 = rule1.clone({ byMonthDay: [3] });
```

### Option 2: Add a `with()` method (similar to Temporal API)
```typescript
const rule2 = rule1.with({ byMonthDay: [3] });
```

### Option 3: Make `options()` return a deep copy
This would prevent accidental mutations but still doesn't solve the cloning issue.

## Additional Context
This library is built on Temporal API which follows immutable patterns. It would be consistent to apply similar principles to `RRuleTemporal` as well.

**Comment by ggaabe** [comment]
See https://github.com/ggaabe/rrule-temporal/pull/92

**Comment by MOhhh-ok** [comment]
Thank you for implementing the `with()` method. I've confirmed that it works perfectly in React.

```tsx
import { Temporal } from "@js-temporal/polyfill";
import { useState } from "react";
import { RRuleTemporal } from "rrule-temporal";

export default function App() {
  const [rule, setRule] = useState<RRuleTemporal>(
    new RRuleTemporal({
      freq: "MONTHLY",
      dtstart: Temporal.ZonedDateTime.from({
        year: 2026,
        month: 1,
        day: 1,
        timeZone: "Asia/Tokyo",
      }),
      byMonthDay: [1],
    }),
  );

  return (
    <div>
      <button
        onClick={() => {
          const newRule = rule.with({
            byMonthDay: [rule.options().byMonthDay![0] + 1],
          });
          // Verify that modifying the original rule doesn't affect newRule
          rule.options().byMonthDay = [1];
          setRule(newRule);
        }}
      >
        TEST
      </button>
      {rule.options().byMonthDay?.join(", ")}
    </div>
  );
}
```
